### ISSUE TITLE
Fix double finalization of detached Vec-backed ArrayBuffer

### ISSUE THREAD

**Issue by yorkie** [jsar-project/rquickjs#9]
## Problem

`ArrayBuffer::new` and `TypedArray::new` transfer a Rust `Vec` allocation to QuickJS for zero-copy access. Their `drop_raw` callback reconstructs that allocation with `Vec::from_raw_parts`.

When the ArrayBuffer is detached, QuickJS calls the callback once with the original data pointer, then sets `abuf->data = NULL`. When the detached ArrayBuffer is later finalized, `js_array_buffer_finalizer` calls the same callback again with the null pointer. The second call attempts to reconstruct a `Vec` from null, violating `Vec::from_raw_parts`' pointer requirements and potentially causing an abort or undefined behavior.

This is present at revision `bccc15d92750c88d12b8724bfbef24b54ba318d0`:

- `core/src/value/array_buffer.rs`: `ArrayBuffer::new` installs `drop_raw`, which unconditionally calls `Vec::from_raw_parts(ptr, capacity, capacity)`.
- `sys/quickjs/quickjs.c`: `JS_DetachArrayBuffer` invokes `abuf->free_func(...)` and sets `abuf->data = NULL`; `js_array_buffer_finalizer` later invokes `abuf->free_func(...)` again.

## Reproduction

```rust
use rquickjs::{ArrayBuffer, Context, Result, Runtime};

fn main() -> Result<()> {
    let runtime = Runtime::new()?;
    let context = Context::full(&runtime)?;

    context.with(|ctx| {
        let mut buffer = ArrayBuffer::new(ctx, vec![1_u8, 2, 3, 4])?;
        buffer.detach();
        drop(buffer);
        Ok(())
    })?;

    runtime.run_gc();
    Ok(())
}
```

The same path can be reached through `TypedArray::new`, because it creates its backing buffer with `ArrayBuffer::new`.

## Expected behavior

A Vec-backed ArrayBuffer can be detached and later garbage-collected while its allocation is released exactly once. `ArrayBuffer::new` and `TypedArray::new` should remain safe zero-copy constructors.

## Suggested direction

Ensure the external buffer callback is called only once, or make the Rust callback retain enough state to recognize that the allocation has already been released. A regression test should cover both explicit `ArrayBuffer::detach()` followed by GC and detachment through JavaScript APIs that transfer an ArrayBuffer.

**Comment by yorkie** [comment]
Fixed by PR #10, which is merged and has passing CI.
